sql/_36_guava/partition_table/cases/cbrd_25519.sql

```diff
@@ -0,0 +1,241 @@
+partition by list (ca)
+(
+  partition p0 values in (0, 1),
+  partition p1 values in (2, 3),
+  partition p2 values in (4, 5),
+  partition p3 values in (6, 7),
+  partition p4 values in (8, 9)
+);
+create table tb (ca int, cb int, cc int, cd int, index i1 (cb, cc), primary key (ca, cd));
+
+insert into tb
+  with recursive cte (n) as (
+      select 0
+      union all
+      select n + 1 from cte where n < 9
+    )
+  select a.n, b.n, c.n, rownum from cte a, cte b, cte c;
+insert into ta_range select * from tb;
+insert into ta_list select * from tb;
+insert into ta_hash select * from tb;
+
+update statistics on ta_range, ta_list, ta_hash, tb with fullscan;
+
+set trace on;
+
+-- START TEST
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. agl');
+
+select /*+ recompile */ count (*)
+from ta_range a
+using index none;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. all partitions (range)');
+
+select /*+ recompile */ *
+from ta_range a
+using index none
+order by cd desc
+limit 1;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. all partitions (hash)');
+
+select /*+ recompile */ *
+from ta_hash a
+using index none
+order by cd desc
+limit 1;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. all partitions (list)');
+
+select /*+ recompile */ *
+from ta_list a
+using index none
+order by cd desc
+limit 1;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning (range)');
+
+select /*+ recompile */ count (*)
+from ta_range a
+where a.ca between 1 and 4;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning (hash)');
+
+select /*+ recompile */ count (*)
+from ta_hash a
+where a.ca in (2, 3);
+show trace;
+
+select /*+ recompile */ count (*)
+from ta_hash a
+where a.ca >= 9;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning (list)');
+
+select /*+ recompile */ count (*)
+from ta_list a
+where a.ca in (5, 6);
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning + limit');
+
+select /*+ recompile */ *
+from ta_range a
+order by cd
+limit 150, 10;
+show trace;
+
+select /*+ recompile */ *
+from ta_range a
+order by cd
+limit 550, 10;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning + inline view + limit');
+
+select /*+ recompile ordered */
+  count(*)
+from
+  (
+    select *
+    from ta_range a
+    where a.ca between 5 and 7
+    limit 250
+  ) a,
+  tb b
+where a.ca = b.ca and a.cd = b.cd;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. partition table + nl join');
+
+select /*+ recompile ordered use_nl */ *
+from ta_range a, ta_hash b, tb c, ta_list d
+where a.ca = b.ca and a.cd = b.cd and b.ca = c.ca and b.cd = c.cd and c.ca = d.ca and c.cd = d.cd
+order by a.cd, a.ca
+limit 250, 10;
```

Comment:
ORDER BY 절이 추가되면 테스트 목적과 달라집니다.
아래처럼 변경해주세요.
```suggestion
select /*+ recompile */ count (*)
from (
    select /*+ ordered use_nl */ a.cd as a_cd, b.cd as b_cd, c.cd as c_cd, d.cd as d_cd
    from ta_range a, ta_hash b, tb c, ta_list d
    where a.ca = b.ca and a.cd = b.cd and b.ca = c.ca and b.cd = c.cd and c.ca = d.ca and c.cd = d.cd
    limit 250, 10
  );
```